api/chat.py

```diff
@@ -0,0 +1,116 @@
+from fastapi import APIRouter
+from api.database import SessionLocal, Memory, simples_distance_query
+from api.utils_open_ai import embed_text
+from api.tools import tools, execute_tool, detect_tool_call
+from api.schemas import ChatRequest
+from openai import OpenAI
+import os
+
+router = APIRouter()
+client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"))
+
+@router.post("/chat")
+async def chat_endpoint(body: ChatRequest):
+    user_msg = body.mensagem
+    query_vector = embed_text(user_msg)
+
+    with SessionLocal() as session:
+        historico_raw = simples_distance_query(session, query_vector)
+        historico = [(m.content, d) for m, d in historico_raw]
+
+    messages = []
+    for content, _ in historico:
+        if content.strip().lower().startswith("assistente:"):
+            messages.append({"role": "assistant", "content": content})
+        else:
+            messages.append({"role": "user", "content": content})
```

Comment:
Como voce só vai utilizar o content dentro do historico, voce poderia ter retornado apenas ele no mapeamento desse jeito aqui

```suggestion
        historico = [m.content for m, d in historico_raw]

    messages = []
    for content, _ in historico:
        if content.strip().lower().startswith("assistente:"):
            messages.append({"role": "assistant", "content": content})
        else:
            messages.append({"role": "user", "content": content})
```